src/sync/impl/sync_file.cpp

```diff
@@ -200,7 +201,15 @@ std::string create_timestamped_template(const std::string& prefix, int wildcard_
     wildcard_count = std::min(WILDCARD_MAX, std::max(WILDCARD_MIN, wildcard_count));
     std::time_t time = std::time(nullptr);
     std::stringstream stream;
-    stream << prefix << "-" << std::put_time(std::localtime(&time), "%Y%m%d-%H%M%S") << "-" << std::string(wildcard_count, 'X');
+    stream << prefix << "-";
+#if __GNUC__ < 5
```

Comment:
I think the limitation comes from the NDK and not from GCC 4.9 itself. Maybe we could add a check for android as well? Same goes for the other changes.